modules/base/rendering/screenspaceinsetblackout.cpp

```diff
@@ -0,0 +1,768 @@
+    std::string strRight = formatLine("Right", rightSplineData);
+    std::string strBottom = formatLine("Bottom", bottomSplineData);
+    std::string strLeft = formatLine("Left", leftSplineData);
+
+    ghoul::setClipboardText(strCorners + strTop + strRight + strBottom + strLeft);
+}
+
+std::string ScreenSpaceInsetBlackout::BlackoutShape::formatLine(std::string id,
+                                                    const std::vector<glm::vec2>& data)
+{
+    std::string str = std::format("{} = {{ ", id);
+    for (int i = 0; i < data.size(); ++i) {
+        std::string xVal = std::format("{}", data[i].x);
+        std::string yVal = std::format("{}", data[i].y);
+        xVal += (xVal.find(".") == std::string::npos) ? ".0" : "";
+        yVal += (yVal.find(".") == std::string::npos) ? ".0" : "";
+        str.append(std::format("{{{}, {}}}", xVal, yVal));
+        std::string delimiter = (i < data.size() - 1) ? ", " : " ";
+        str.append(delimiter);
+    }
+    str.append("},\n");
+
+    return str;
+}
+
+ScreenSpaceInsetBlackout::ScreenSpaceInsetBlackout(const ghoul::Dictionary& dictionary)
+    : ScreenSpaceRenderable(dictionary)
+    , _blackoutShape(dictionary)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    std::string identifier = p.identifier.value_or("ScreenSpaceInsetBlackout");
+    setIdentifier(makeUniqueIdentifier(std::move(identifier)));
+
+    // Makes sure that User has specified 4 corners in the asset file
+    checkCornerSpecification(p.blackoutshape.corners);
+
+    // Add to GUI
+    addPropertySubOwner(_blackoutShape);
+
+    // Handling of calibration texture
+    std::optional<std::string> optTexturePath = p.blackoutshape.calibrationTexturePath;
+    if (optTexturePath.has_value()) {
+        if (std::filesystem::is_regular_file(absPath(*optTexturePath))) {
+            std::filesystem::path path = absPath(*optTexturePath);
+            std::unique_ptr<ghoul::opengl::Texture> texture =
+                ghoul::io::TextureReader::ref().loadTexture(path, 2);
+            if (texture) {
+                // Images don't need to start on 4-byte boundaries, for example if the
+                // image is only RGB
+                glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
+
+                if (texture->format() == ghoul::opengl::Texture::Format::Red) {
+                    texture->setSwizzleMask({ GL_RED, GL_RED, GL_RED, GL_ONE });
+                }
+
+                texture->uploadTexture();
+                texture->setFilter(ghoul::opengl::Texture::FilterMode::LinearMipMap);
+                texture->purgeFromRAM();
+
+                _calibrationTexture = std::move(texture);
+                _objectSize = _calibrationTexture->dimensions();
+            }
+        }
+        else {
+            LWARNINGC(
+                "ScreenSpaceInsetBlackout",
+                std::format(
+                    "Calibration texture '{}' is missing.",
```

Comment:
```suggestion
                    "Could not find calibration texture '{}'.",
```

"is missing" made me think it was required :)